hamilton/execution/graph_functions.py

```diff
@@ -285,6 +251,84 @@ def dfs_traverse(
     return computed
 
 
+# TODO: better function name
+def execute_lifecycle_for_node(
```

Comment:
We should probably have some dedicated unit tests? I know existing tests probably cover most of it. You could have a look at code coverage to see if any lines are missed?